text/0000-rustdoc-types-maintainers.md

```diff
@@ -0,0 +1,134 @@
+
+Republishing `rustdoc-json-types` as `rustdoc-types` is done with [a script](https://github.com/aDotInTheVoid/rustdoc-types/blob/17cbe9f8f07de954261dbb9536c394381770de7b/update.sh) so that it is as low maintenance as possible. This also ensures that all format/documentation changes happen in the rust-lang/rust repo, and go through the normal review process there.
+
+The update/publishing process will be moved to T-rustdoc. In the medium term, I (`@aDotInTheVoid`) will still do it, but
+- In an official capacity
+- With bus factor for when I stop.
+
+We (T-rustdoc) will continue to publish a new version of the `rustdoc-types` crate
+every time the upstream implementation changes, and these will be versioned with
+normal SemVer. Changes to rustdoc-json in `rust-lang/rust` will not be accepted
+if they would make it not possible to publish `rustdoc-types`.
+
+## Actual Mechanics of the move
+
+### GitHub
+
+GitHub has a [list of requirements](https://docs.github.com/en/repositories/creating-and-managing-repositories/transferring-a-repository) for transferring repositories. T-infra will handle the permissions of moving the repository into the rust-lang GitHub organization.
+
+At the end of this we should have a moved the [`aDotInTheVoid/rustdoc-types`]
+repo into the rust-lang GitHub org. T-rustdoc will have `maintain` permissions
+(via the [team repo](https://github.com/rust-lang/team/)).
+
+### crates.io
+
+crates.io ownership is managed [via the command line](https://doc.rust-lang.org/cargo/reference/publishing.html#cargo-owner).
+
+I will run the following commands to move ownership.
+
+```
+cargo owner --add github:rust-lang:rustdoc
+cargo owner --add rust-lang-owner
+cargo owner --remove aDotInTheVoid
+```
+
+The `rust-lang-owner` is needed because team owners cannot add new owners. 
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+- Adds additional maintenance burden to rustdoc team.
+- One-time maintenance burden to infra team to support move.
+
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+- We could keep `rustdoc-types` as a personal project. This preserves the status quo (and is what will happen if this RFC (or something similar) isn't adopted). This is undesirable because
+      - Bus factor: If I am unable or unwilling to maintain `rustdoc-types`, we cause a load of unnecessary churn when it becomes out of sync with the in-tree `rustdoc-json-types`
+- We could bundle `rustdoc-types` through rustup. This is undesirable as it means users can't depend on it in stable rust, and can't depend on multiple versions.
+- We could publish `rustdoc-json-types` directly from `rust-lang/rust`. However
```

Comment:
Hmm, I am a little concerned about having a separate repo because it would mean every PR that increases FORMAT_VERSION would also necessitate a separate PR to a different repo. Is there a downside to publishing from a folder in rust-lang/rust instead (or maybe even a git subtree)? See also my comments below about merging it with rustdoc-json-types, though my main concern is requiring multiple PRs.

Reply:
> though my main concern is requiring multiple PRs.

FWIW, this is how it's always been done, but that's defiantly not sufficient justification that it's the best way.

> Is there a downside to publishing from a folder in rust-lang/rust instead (or maybe even a git subtree)?

1. Can't tag the git repo with creates.io version
    - I'm not sure how compelling this is
2. No way to diff view diff between releases
    - https://github.com/aDotInTheVoid/rustdoc-types/compare/v0.29.1...v0.30.0 is nice to see.
    - Know way (AFAIKT) to get something like this in rl/r (without github showing changes from the rest of the repo)
3. Having the crate be a small repo makes it easier to depend on it via cargo's git feature (rather than having cargo clone rl/r). https://github.com/aDotInTheVoid/rustdoc-types/pull/14
4. Requires re-engineering release procedures, and it's unclear how that would work in rust-lang/rust
     - Are we now going to cd into a local clone of rust-lang/rust and `cargo publish`? This is involved to
     - Alternatively, is rust-lang/rust CI going to autopublish? This is a whole can of worms, and it'd be much easier to set up auto-publishing on it's own repo. 
     - Changelog generation get's tricky if it needs to be done in the PR that implements it.
         - Can't know the date it will be published to crates.io before merging, due to unpredictable bors delay
             - But maybe we don't need this in the changelog, it's not super intuitive how this is ties to _crates.io_ (not rustup dates)
5. I kinda like that the publish happens later, and gets another pass on it before going out to users:
    - A couple of big sweeping compiler changes have made breaking changes to rustdoc-json. (rust-lang/rust#125958, rust-lang/rust#120131). Knowing that as long as FORMAT_VERSION was bumped we'd be fine, (as everything else could be hashed out separately later) made this less stressful for me, and reduced workload on T-Compiler
    - But maybe this is just me wanting to keep control of things, and want to hoard responsibility withing T-Rustdoc. It feels kinda unhealthy somehow.

Reply:
Whats your concern with multiple PR's? Given that there's already a publish step, I don't think it cuts down on work (unless we autopublish from rust-lang/rust PR's, which I'm not fully comfortable with)

We could land this now (primarily for bus-factor reasons), and then move change things separately if it becomes a problem.

Reply:
> Whats your concern with multiple PR's? Given that there's already a publish step, I don't think it cuts down on work (unless we autopublish from rust-lang/rust PR's, which I'm not fully comfortable with)

Fair enough since all that's needed to update the repo is to run a script.

> We could land this now (primarily for bus-factor reasons), and then move change things separately if it becomes a problem.

Yeah, I guess that makes sense since it wouldn't even need a whole RFC for a small administrative change like that.

Reply:
> I kinda like that the publish happens later, and gets another pass on it before going out to users:

Ah, the impression I got from the RFC text was you wanted publishing to happen almost as soon as the format version changed. I agree that there's benefit to reviewing the changes before publishing a new crate version.